### ISSUE TITLE
Wax+Leaflet mouse position falls out of out of sync in Mac Chrome after zoom/click/drag events

### ISSUE THREAD

**Issue by palewire** [cutting-room-floor/wax#266]
I'm encountering a bug, only in Mac versions of Chrome so far. Here's what happens. 

I'm calling in a Wax interaction layer for a set of polygons.  When I load the page and hover over them, it works great and I can render accurate popups about the selected features.

But... if a Mac Chrome user zooms in while in the middle of rapidly moving the cursor around the page, the map zooms but then the mouse position is suddenly offset to a spot far away from the actual mouse position. The offset is consistently wrong, so as you continue to move your mouse around, it lights up areas the same X and Y pixels away from your true position.

This issue also occurs in Mac Chrome when users click and drag while moving around the map.

I am using Leaflet version 0.4.5 and the latest version of Wax.

I do not have this issue in my Linux Chromium.

**Comment by palewire** [comment]
You can see the bug in action on this map. http://graphics.latimes.com/how-la-voted-2012/

**Comment by palewire** [comment]
On @miketahani advice I upgraded to the trunk wax.leaflet file and some of my Mac testers are reporting slightly better, though still buggy, performance.

**Comment by tmcw** [comment]
Are they still seeing this particular bug after the update? I cannot replicate in Mac Chrome Beta.

**Comment by tmcw** [comment]
Also, looks like there was a removed comment to the effect of:

> one thing that isn't quite a bug but is still pretty annoying: all the 
> tiles have to load before the UTFgrid data is loaded, so if there's a 
> hiccup in the connection to the tileserver or something and a tile gets 
> delayed, the whole page will hang and interactivity will cease until that 
> tile is finished

This isn't how Wax works; you're either seeing how Leaflet works or using custom code. Wax loads grids on mousemove over tiles, and unless Leaflet provides a more fine-grained event, listens for tilesloaded.

**Comment by miketahani** [comment]
yeah, sorry tom, that wasn't meant for the issue tracker. i got an email
notification and i hit reply without looking at where it was going. if
you're still seeing it somewhere, please feel free to delete it.

thanks,
mike

**Comment by palewire** [comment]
@tmcw They reported improvement with the trunk, though I haven't been able to test it personally. I don't know what the actual cause is, but the result is so crippling for that subset of users that I would summon up the arrogance to suggest you put out an official release with the fix ASAP.

**Comment by tmcw** [comment]
Sure, it's been a while since a tag, tagged dev12.

**Comment by kevinschaul** [comment]
This is still very buggy. Anyone know of remedies? Using `6.4.2 - v6.0.4-31-g769f4ce`

**Comment by kevinschaul** [comment]
This crippling bug also exists in the latest Chrome on Ubuntu 12.10.

**Comment by tmcw** [comment]
@kevinschaul you might want to use https://github.com/danzel/Leaflet.utfgrid or, if you can pull together a fix, submitting a pull request.

**Comment by Abe808** [comment]
Go back to basics. Check if a page with just your map, nothing else, has the problem.

After having the same problem across multiple browsers I checked the first page I created that only had the map nothing else. My mouse position was correct. Hence, there was no issue with Leaflet.

Further searching led me to a similar issue with JQuery (http://forum.jquery.com/topic/jquery-reporting-wrong-x-y-positions). This helped me to solve my problem."The popup will not get an absolute position relative to the page when it's parent element has position:relative, it will get absolute positioned relative to that position:relative parent" [rdworth].

I simply created a  div element around my map, assigned it "position:absolute; left:400px;" and the map appeared in the right spot and now the mouse coordinates were back in sync with the mouse position.
